### PR TITLE
Add failing test fixture for AddReturnTypeDeclarationRector

### PR DESCRIPTION
**Pull request by mvhirsch**
# Failing Test for AddReturnTypeDeclarationRector

Based on https://getrector.org/demo/77038c39-315d-4807-81b7-915116e935a5

refs [#7253](https://github.com/rectorphp/rector/issues/7253)

### CODE REVIEW

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/Fixture/my_form_type.php.inc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Symfony\Component\Form;
+
+interface FormTypeInterface
+{
+    /**
+     * @return string|null
+     */
+    public function getParent();
+}
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/Fixture/my_form_type.php.inc]: You can move interface to `Source` directory https://github.com/rectorphp/rector-src/tree/main/rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/Source

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/Fixture/my_form_type.php.inc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddReturnTypeDeclarationRector\Fixture;
+
+final class MyFormType implements FormTypeInterface
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/Fixture/my_form_type.php.inc]: You need to register `FormTypeInterface` to config https://github.com/rectorphp/rector-src/blob/main/rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/config/configured_rule.php

**Comment by mvhirsch** [comment]
moved and registered classes/rules. like this, @samsonasik?

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/config/configured_rule.php [hunk 3]
```diff
@@ -24,5 +29,6 @@
                 new SimpleStaticType(ReturnOfStatic::class)
             ),
             new AddReturnTypeDeclaration(DataTransformerInterface::class, 'transform', new MixedType()),
+            new AddReturnTypeDeclaration(FormTypeInterface::class, 'getParent', $nullableStringType),
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddReturnTypeDeclarationRector/config/configured_rule.php:33]: it need to target to `Source` path location:

```suggestion
            new AddReturnTypeDeclaration(\Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddReturnTypeDeclarationRector\Source\FormTypeInterface::class, 'getParent', $nullableStringType),
```

**Comment by samsonasik** [comment]
@mvhirsch I cherry-picked your commit at PR https://github.com/rectorphp/rector-src/pull/2559

**Pull request closed without merging**